evolver/base.py

```diff
@@ -42,29 +144,56 @@ class Config(BaseConfig):
         ...
 
     @classmethod
-    def create(cls, config=None):
+    def create(cls, config: ConfigDescriptor | dict | str | None = None):
         """ Create an instance from a config. """
+
+        def validate_descriptor(descriptor: ConfigDescriptor):
+            if not issubclass(cls, descriptor.classinfo):
+                raise TypeError(f"The given {ConfigDescriptor.__name__} for '{descriptor.classinfo}' is not compatible "
+                                f"with this class '{cls.__qualname__}'")
```

Comment:
🤔 Perhaps.

``model_validate(..., context=dict(extra="forbid"))`` as we have would allow for the model validation to fail as we'd want for that to work - so that's covered.

As it is in this PR, inheritance is still allowed since it's checking on ``issubclass``. So not checking would imply a legitimate use where the classes weren't inherited (as you might expect them to properly be) _and_ the configs _just_ happen to perfectly match one another.

I think not checking as is in this PR, hinges the type/inheritance correctness entirely on the ``Config``, which I don't think is very robust - it seems too indirect.

Reply:
Wouldn't letting whatever error propogate from below be more direct? As it is now we state they are not compatible, but in fact they could be and if they weren't it is probably because Config are not compatible and an error from there would be more direct (e.g. it might say which field is missing, etc)?

Anyway, I think it is OK here, just was wondering

Reply:
Yeah, I get that they _could_ be compatible by ``Config``, however, they may not be by other paths, i.e., func methods and other attributes etc. The things that inheritance gives you by design.

Also, whilst they may match on config field names, they may not match on defaults, nor types (since we don't use ``strict=True``), nor any other aspects that they could differ by.

We shouldn't ignore the other option, that perhaps we shouldn't allow any of this, i.e., subclasses aren't even allowed and to ``create`` from a descriptor requires explicitly for ``classinfo`` to match that of the ``cls.create()`` being called. This was going to be my default stance, however, I felt that too strict and that subclassing should allow for descriptors to be re-used. Even, then, the subclass's config may add new required fields that prevent even that - resulting in a validation error as you stated.